WooCommerce/WooCommerceTests/ViewRelated/Orders/Order Creation/Synchronizer/RemoteOrderSynchronizerTests.swift

```diff
@@ -727,6 +1062,44 @@ final class RemoteOrderSynchronizerTests: XCTestCase {
         XCTAssertEqual(states, [.syncing(blocking: false), .synced])
     }
 
+    func test_states_are_properly_set_upon_success_order_update_via_productID_input_with_no_new_items() {
+        // Given
+        let stores = MockStoresManager(sessionManager: .testingInstance)
+        let synchronizer = RemoteOrderSynchronizer(siteID: sampleSiteID, flow: .creation, stores: stores)
+
+        stores.whenReceivingAction(ofType: OrderAction.self) { action in
+            switch action {
+            case .createOrder(_, _, let completion):
+                completion(.success(.fake().copy(orderID: self.sampleOrderID)))
+            case .updateOrder(_, let order, _, let completion):
+                completion(.success(order))
+            default:
+                XCTFail("Unexpected action: \(action)")
+            }
+        }
```

Comment:
When you have a lot of very similar set up code like this, it can make tests more readable to extract it to a helper function which surfaces the key information about what you're arranging, and reduces the duplication. It also makes the tests easier to change, potentially.

See my [recent changes to the PaymentMethodsViewModel for an example](https://github.com/woocommerce/woocommerce-ios/blob/8a3fc213f662dc3badbe1505e6ddd9ce45f46266/WooCommerce/WooCommerceTests/ViewRelated/Orders/Payment%20Methods/PaymentMethodsViewModelTests.swift#L748) – it's not a widespread practice in our code, but it would work well here I think.